File: leetcode/hot100/codes/Go/颜色分类.go
func sortColors(nums []int) {
	ptr_0 := 0
	ptr_2 := len(nums) - 1

	for i := 0; i <= ptr_2; i++ {
		for i < ptr_2 && nums[i] == 2 {
			nums[i], nums[ptr_2] = nums[ptr_2], nums[i]
			ptr_2 -= 1
		}
		if nums[i] == 0 {
			nums[i], nums[ptr_0] = nums[ptr_0], nums[i]
			ptr_0 += 1
		}
	}
}

File: leetcode/hot100/codes/Python/颜色分类.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        ptr_0 = 0
        ptr_2 = len(nums) - 1

        for i in range(ptr_2 + 1):
            while i < ptr_2 and nums[i] == 2:
                nums[i], nums[ptr_2] = nums[ptr_2], nums[i]
                ptr_2 -= 1
            if nums[i] == 0:
                nums[i], nums[ptr_0] = nums[ptr_0], nums[i]
                ptr_0 += 1
        